### PR TITLE
fix: Update SDK so HubPoolClient.batchCompute handles multiple origin chains

### PR DESCRIPTION
**Pull request by nicholaspai**
I also added a test for this scenario

### CODE REVIEW

**Review by pxrl — APPROVED** [review]
LGTM.

File: src/dataworker/DataworkerUtils.ts [hunk 1]
```diff
@@ -129,7 +129,7 @@ export async function blockRangesAreInvalidForSpokeClients(
     }
 
     if (endBlockTimestamps !== undefined) {
-      assert(Object.keys(endBlockTimestamps).length === blockRanges.length);
+      assert(Object.keys(endBlockTimestamps).length === Object.values(spokePoolClients).filter(isDefined).length);
```
Comment (james-a-morris) [inline src/dataworker/DataworkerUtils.ts:132]: +1. But shouldn't we already filter `null/undefined` `spokePoolClient` instances?

**Review by james-a-morris — APPROVED** [review]
LGTM - one comment

File: src/dataworker/DataworkerUtils.ts (hunk 1 above)
Comment (nicholaspai) [reply src/dataworker/DataworkerUtils.ts]: yeah but that's not what we're checking. Anyways I moved this check up higher https://github.com/across-protocol/relayer-v2/pull/1208/commits/3671dd82e9c6b97f8cd2afe70d32c03a3291fab6

**Pull request merged**